### PR TITLE
[mono] More domain cleanups.

### PR DESCRIPTION
**Pull request by vargaz**
Based on https://github.com/dotnet/runtime/pull/48023.

### CODE REVIEW

File: src/mono/mono/metadata/mono-mlist.c [hunk 1]
```diff
@@ -69,7 +69,7 @@ mono_mlist_alloc_checked (MonoObject *data, MonoError *error)
 	MonoMList* res;
 	if (!monolist_item_vtable) {
 		MonoClass *klass = mono_class_load_from_name (mono_defaults.corlib, "Mono", "MonoListItem");
-		monolist_item_vtable = mono_class_vtable_checked (mono_get_root_domain (), klass, error);
+		monolist_item_vtable = mono_class_vtable_checked (klass, error);
```
Comment (lambdageek) [inline src/mono/mono/metadata/mono-mlist.c:72]: mono-mlist.[ch] aren't in the CMakeLists.txt anymore. I think the files can be removed

File: src/mono/mono/metadata/object-internals.h [hunk 2]
```diff
@@ -1490,7 +1468,7 @@ char *
 mono_string_to_utf8_image (MonoImage *image, MonoStringHandle s, MonoError *error);
 
 MonoArrayHandle
-mono_array_clone_in_domain (MonoDomain *domain, MonoArrayHandle array, MonoError *error);
+mono_array_clone_in_domain (MonoArrayHandle array, MonoError *error);
```
Comment (lambdageek) [inline src/mono/mono/metadata/object-internals.h:1471]: TODO: replace all uses by `mono_array_clone_checked`, I guess

File: src/mono/mono/metadata/object-internals.h [hunk 3]
```diff
@@ -1797,34 +1765,34 @@ gboolean
...
-mono_ldstr_checked (MonoDomain *domain, MonoImage *image, uint32_t str_index, MonoError *error);
+mono_ldstr_checked (MonoImage *image, uint32_t str_index, MonoError *error);
 
 MonoStringHandle
-mono_ldstr_handle (MonoDomain *domain, MonoImage *image, uint32_t str_index, MonoError *error);
+mono_ldstr_handle (MonoImage *image, uint32_t str_index, MonoError *error);
 
 MONO_PROFILER_API MonoString*
-mono_string_new_checked (MonoDomain *domain, const char *text, MonoError *merror);
+mono_string_new_checked (const char *text, MonoError *merror);
```
Comment (lambdageek) [inline src/mono/mono/metadata/object-internals.h:1783]: There's a couple uses in profiler/coverage.c

File: src/mono/sample/HelloWorld/Makefile [hunk 4]
```diff
@@ -2,7 +2,7 @@ TOP=../../../../
 DOTNET:=$(TOP)./dotnet.sh
 DOTNET_Q_ARGS=--nologo -v:q -consoleloggerparameters:NoSummary
 
-MONO_CONFIG ?=Release
+MONO_CONFIG ?=Debug
```
Comment (lambdageek) [inline src/mono/sample/HelloWorld/Makefile:5]: nit: unrelated.  but it also doesn't matter

**Review by lambdageek — APPROVED** [review]

File: src/mono/sample/HelloWorld/Makefile (hunk 4 above)
Comment (vargaz) [reply src/mono/sample/HelloWorld/Makefile:5]: Will address in a following PR.

**Pull request merged**